src/storyMap/components/StoryMap.js

```diff
@@ -190,24 +185,95 @@ const getTransition = (config, id) => {
   };
 };
 
-const StoryMap = props => {
-  const { t } = useTranslation();
-  const {
-    config,
-    onStepChange,
-    ChapterComponent = Chapter,
-    TitleComponent = Title,
-    mapCss = { height: '100vh', width: '100vw', top: 0 },
-    animation,
-    onReady,
-    chaptersFilter,
-  } = props;
-  const mapContainer = React.useRef(null);
-  const mapInsetContainer = React.useRef(null);
-  const [map, setMap] = React.useState(null);
-  const [insetMap, setInsetMap] = React.useState(null);
-  const [marker, setMarker] = React.useState(null);
-  const [isReady, setIsReady] = React.useState(false);
+const InsetConfig = props => {
+  const { map, insetMap } = props;
+
+  useEffect(() => {
+    if (!map || !insetMap) {
+      return;
+    }
+
+    function addInsetLayer(bounds) {
+      insetMap.addSource('boundsSource', {
+        type: 'geojson',
+        data: bounds,
+      });
+
+      insetMap.addLayer({
+        id: 'boundsLayer',
+        type: 'fill',
+        source: 'boundsSource', // reference the data source
+        layout: {},
+        paint: {
+          'fill-color': theme.palette.white,
+          'fill-opacity': 0.2,
+        },
+      });
+      // Add a black outline around the polygon.
+      insetMap.addLayer({
+        id: 'outlineLayer',
+        type: 'line',
+        source: 'boundsSource',
+        layout: {},
+        paint: {
+          'line-color': theme.palette.black,
+          'line-width': 1,
```

Comment:
question: should this be in a constant or configuration variable?

Reply:
I don't think this belongs in a constant or configurable value, this is similar to the inline styling we have for any component, we don't have constant for that, and also this is not being reused